### PR TITLE
Process function scope variable annotations

### PR DESCRIPTION
**Pull request by jstasiak**
Even though variable annotations in function scope aren't evaluated at
runtime it's still useful for static analysis tools to process them and
catch some issues (and not report some things that aren't issues).

Let's take the following code:

```python
     from typing import Any

     def fun():
          a: Any
```

Previously pyflakes would report Any to be unused:

```
     test.py:1: 'typing.Any' imported but unused
```

With this patch it's no longer the case.

### CODE REVIEW

**Comment by jstasiak** [comment]
I'm not sure about those CI tasks being stuck for hours now but as far as I know the CI suite passes with this change.

**Comment by sigmavirus24** [comment]
@jstasiak AppVeyor is suffering from an outage at the moment. That said, the tests still need to pass there for this to be accepted.

**Comment by jstasiak** [comment]
All the tests passed, let me know if there are any modifications I need to make here.

**Comment by ambv** [comment]
Instead of commenting out why ANNASSIGN is a separate method, you could have deleted it entirely.

**Comment by sigmavirus24** [comment]
@ambv Care to send a PR to that effect?

**Comment by ambv** [comment]
Actually, nevermind, we still need it as the first check is still custom: `if node.value: handle node.target`.

**Comment by asmeurer** [comment]
Function scope annotations are ignored completely (so long as they're valid Python). With this PR, you'd get

```py
def test():
    a: fdsf = 1
    return a
print(test())
```

```
test.py:2: undefined name 'fdsf'
```

even though if you run the program, it prints `1`.

It probably makes sense to allow names used in function annotations to be imported without unused warnings, as explicitly importing `Any` from `typing` serves as documentation. But should it fail in the above case? I really don't know what the best practice is, but in a technical sense, this *is* a false positive.

**Comment by ambv** [comment]
Yeah, this is why originally I put the special case for function-scope annotations. However, the unused import warning is clearly disruptive and I changed my mind. What @jstasiak did here is right.

Moreover, the most likely cause of an invalid function-scope annotation will be a typo. The user *wants* to see a warning about it.

The reason that function-scope annotations aren't evaluated is purely efficiency. They would need to be evaluated on every call which is prohibitively costly. The idea to instead "evaluate them on function creation" was rejected by Guido because their presence inside a function clearly suggests they are in function scope. They can also successfully used function-local imports.

Summing up, @asmeurer, I wouldn't worry about the "technically" false positive here. We are doing the right thing now.

**Comment by dahlia** [comment]
When would this fix be released to PyPI?

**Comment by ambv** [comment]
It is part of 1.5.0.

**Comment by Daenyth** [comment]
I'm using 1.5.0 and still experiencing this error.

```
$ pip freeze | grep pyflakes
pyflakes==1.5.0

$ cat foo.py
from typing import Any
def foo(x):
    y: Any = x
    return y

$ python -m pyflakes foo.py
foo.py:1: 'typing.Any' imported but unused
```

**Comment by myint** [comment]
1.5.0 was from 9 January 2017. This pull request was merged in 22 January 2017.

**Comment by Daenyth** [comment]
Ok, I misunderstood what @ambv said to mean that the fix was in 1.5.0
